supplier_app/views.py

```diff
@@ -270,13 +249,24 @@ def company_invite(request):
 
 
 @permission_required('users_app.supplier_role', raise_exception=True)
-def company_join(request, token):
-    user = request.user
-    companyuniquetoken = get_object_or_404(CompanyUniqueToken, token=token)
-    company = companyuniquetoken.company
-    CompanyUserPermission.objects.create(user=user, company=company)
-    companyuniquetoken.delete()
-    return redirect('supplier-home')
+def company_join(request, *args, **kwargs):
+    companyuniquetoken = _get_companyuniquetoken_from_token(kwargs['token'])
+    if _token_is_valid(companyuniquetoken):
+        user = request.user
+        company = companyuniquetoken.company
+        CompanyUserPermission.objects.create(user=user, company=company)
+        companyuniquetoken.delete()
+        return redirect('supplier-home')
+    return HttpResponseRedirect(Http404)
+
+
+def _token_is_valid(companyuniquetoken):
+    if companyuniquetoken and not companyuniquetoken.is_token_expired:
+        return True
```

Comment:
You could also return that if condition. Example:
```
def _token_is_valid(company_unique_token):
    return company_unique_token and not company_unique_token.is_token_expired
```

Reply:
done